### PR TITLE
Only registered users can display full article

### PR DESCRIPTION
**Pull request by Fraseer**
PT Link: https://www.pivotaltracker.com/story/show/179938558

## User Story
As a news site
In order to display the full articles to only registered users
I want to restrict visitors from reading the articles

## Changes Proposed
- Create state for registered status in application state
- Add functionality to `rootReducer` and `configureStore` files
- Use `useSelector` on article page get authorised state
- Use ternary operator to prompt visitor to sign in

## Screenshots
image
image

### CODE REVIEW

File: cypress/integration/userCanReadAnArticle.feature.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ describe("User can read an article", () => {
   });
 
   it("is expected to display the article", () => {
+    cy.window().its("store").invoke("dispatch", {
+      type: "SET_CURRENT_USER",
+      payload: true,
+    });
```
Comment (dorianbuck) [inline cypress/integration/userCanReadAnArticle.feature.js:21]: This is noice!!

File: src/components/IndividualArticle.jsx [hunk 2]
```diff
@@ -16,11 +17,37 @@ const IndividualArticle = () => {
...
+            </h2>
+            <ul>
+              Benfits to registering:
+              <li>Full access to our unique, groundbreaking journalism.</li>
+              <li>Read and write comments about the articles you love.</li>
+            </ul>
+            <Button data-cy="register-button" color="orange">
+              Register Now
+            </Button>
+            <>
```
Comment (dorianbuck) [inline src/components/IndividualArticle.jsx]: Is this a must?

File: src/components/IndividualArticle.jsx [hunk 3]
```diff
@@ -16,11 +17,37 @@ const IndividualArticle = () => {
...
+              <div>Already signed up?</div>
+              <Button data-cy="sign-in-button" color="orange">
+                Sign in
+              </Button>
+            </>
+          </Card>
+        </div>
+      ) : (
+        <>
```
Comment (dorianbuck) [inline src/components/IndividualArticle.jsx:41]: same here?

File: cypress/integration/userCanReadAnArticle.feature.js [hunk 4]
```diff
@@ -37,4 +41,13 @@ describe("User can read an article", () => {
     cy.get("[data-cy=home]").click();
     cy.url().should("eq", "http://localhost:3000/");
   });
+
+  describe("when the user in not signed in", () => {
+    it("is expected to display a paywall when not signed in", () => {
```
Comment (maxarvid) [inline cypress/integration/userCanReadAnArticle.feature.js]: Maybe just `is expected to display a paywall`, the user not being signed in is mentioned on 45

File: src/components/IndividualArticle.jsx (hunk 2 above)
Comment (Fraseer) [reply src/components/IndividualArticle.jsx]: do you mean necessary?

File: src/components/IndividualArticle.jsx (hunk 3 above)
Comment (Fraseer) [reply src/components/IndividualArticle.jsx:41]: yarp

File: cypress.json [hunk 5]
```diff
@@ -1,4 +1,6 @@
 {
   "baseUrl": "http://localhost:3000",
-  "chromeWebSecurity": false
+  "chromeWebSecurity": false,
+  "viewportWidth": 350,
```
Comment (maxarvid) [inline cypress.json]: Maybe get rid of these.

**Review by maxarvid — APPROVED** [review]
Mergetastic with the suggested changes

**Review by dorianbuck — APPROVED** [review]
Swell!

File: src/state/reducers/rootReducer.js [hunk 6]
```diff
@@ -1,10 +1,10 @@
 const rootReducer = (state, action) => {
   switch (action.type) {
-    case 'SET_CATEGORIES':
+    case "SET_CATEGORIES":
       return {
         ...state,
-        categories: action.payload
-      }
+        categories: action.payload,
+      };
```
Comment (lukeperrin10) [inline src/state/reducers/rootReducer.js:7]: still formatting issues? can we please fix this?

**Review by lukeperrin10 — APPROVED** [review]
I'll merge this, but you've got to fix the formatters guys. It may not seem like a big thing, but it is a super easy fix.

File: src/state/reducers/rootReducer.js (hunk 6 above)
Comment (Fraseer) [reply src/state/reducers/rootReducer.js:7]: I think this comes from having code merged that hasn't been formatted

**Pull request merged**